hypertrace-graphql-explorer-schema/src/main/java/org/hypertrace/graphql/explorer/dao/GatewayServiceExploreRequestBuilder.java

```diff
@@ -58,23 +59,31 @@ Single<org.hypertrace.gateway.service.v1.explore.ExploreRequest> buildRequest(
         this.filterConverter.convert(request.filterArguments()),
         this.buildAnyAggregations(request),
         this.buildAnyTimeAggregations(request),
-        (attributes, orderBys, groupBys, filter, aggregations, series) ->
-            org.hypertrace.gateway.service.v1.explore.ExploreRequest.newBuilder()
-                .setContext(request.scope())
-                .setStartTimeMillis(request.timeRange().startTime().toEpochMilli())
-                .setEndTimeMillis(request.timeRange().endTime().toEpochMilli())
-                .addAllSelection(attributes)
-                .addAllSelection(aggregations)
-                .addAllTimeAggregation(series)
-                .addAllOrderBy(orderBys)
-                .addAllGroupBy(groupBys)
-                .setLimit(request.limit())
-                .setIncludeRestGroup(request.includeRest())
-                .setOffset(request.offset())
-                .setFilter(filter)
-                .setSpaceId(request.spaceId().orElse("")) // String proto default value
-                .setGroupLimit(request.groupLimit().orElse(0)) // Int proto default value
-                .build());
+        (attributes, orderBys, groupBys, filter, aggregations, series) -> {
+          Builder builder =
+              org.hypertrace.gateway.service.v1.explore.ExploreRequest.newBuilder()
+                  .setContext(request.scope())
+                  .addAllSelection(attributes)
+                  .addAllSelection(aggregations)
+                  .addAllTimeAggregation(series)
+                  .addAllOrderBy(orderBys)
+                  .addAllGroupBy(groupBys)
+                  .setLimit(request.limit())
+                  .setIncludeRestGroup(request.includeRest())
+                  .setOffset(request.offset())
+                  .setFilter(filter)
+                  .setSpaceId(request.spaceId().orElse("")) // String proto default value
+                  .setGroupLimit(request.groupLimit().orElse(0)); // Int proto default value
+          request
+              .timeRange()
+              .ifPresent(
+                  timeRangeArgument -> {
+                    builder.setStartTimeMillis(timeRangeArgument.startTime().toEpochMilli());
+                    builder.setEndTimeMillis(timeRangeArgument.startTime().toEpochMilli());
```

Comment:
Shouldn't this be `timeRangeArgument.endTime().toEpochMilli()`?

Reply:
Wow good eye @skjindal93